hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/offset/kafka/broker/BrokerOffsetsRepository.java

```diff
@@ -69,15 +69,20 @@ public BlockingChannel load(Subscription key) {
 
     public void save(Subscription subscription, PartitionOffset partitionOffset) throws ExecutionException {
         OffsetCommitRequest commitRequest = createCommitRequest(subscription, partitionOffset);
-        OffsetCommitResponse commitResponse = commitOffset(subscription, commitRequest);
-
+        OffsetCommitResponse commitResponse;
+        try {
+            commitResponse = commitOffset(subscription, commitRequest);
+        } catch (Exception e) {
+            channels.invalidate(subscription);
+            throw e;
+        }
         if (commitResponse.hasError()) {
-            commitResponse.errors().values()
-                    .stream()
-                    .map(e -> Short.parseShort(e.toString()))
-                    .filter(e -> (e == ErrorMapping.NotCoordinatorForConsumerCode() || e == ErrorMapping.NotCoordinatorForConsumerCode()))
-                    .forEach(e -> channels.invalidate(subscription));
-
+            for (Object error: commitResponse.errors().values()) {
```

Comment:
But there's no point for that. I want to check if an error of certain type occurred and break the loop closing the channel only once.